### PR TITLE
Dt 1271 smoke test data rest

### PR DESCRIPTION
**Pull request by andymarke**
The purpose of this PR is to provide and endpoint that would allow a test client to "reset" custody data as if the prison-to-probation-update had never been called. This involves

- removing the NOMS number and book number
- setting the prison location back to unknown
- setting the custodial status back to "Sentenced"
- removing the key dates

To also help a test client new endpoints have been added (which are generally useful) which can help to assert that data has been successful reset and set for a successful test

### CODE REVIEW

File: src/test/java/uk/gov/justice/digital/delius/service/CustodyServiceTest.java [hunk 1]
```diff
@@ -43,17 +48,17 @@
...
-    private ConvictionService convictionService = mock(ConvictionService.class);
-    private InstitutionRepository institutionRepository = mock(InstitutionRepository.class);
-    private CustodyHistoryRepository custodyHistoryRepository = mock(CustodyHistoryRepository.class);
-    private ReferenceDataService referenceDataService = mock(ReferenceDataService.class);
-    private SpgNotificationService spgNotificationService = mock(SpgNotificationService.class);
-    private ArgumentCaptor<CustodyHistory> custodyHistoryArgumentCaptor = ArgumentCaptor.forClass(CustodyHistory.class);
-    private OffenderManagerService offenderManagerService = mock(OffenderManagerService.class);
-    private ContactService contactService = mock(ContactService.class);
-    private OffenderPrisonerService offenderPrisonerService = mock(OffenderPrisonerService.class);
+    private final TelemetryClient telemetryClient = mock(TelemetryClient.class);
```
Comment (andymarke) [inline src/test/java/uk/gov/justice/digital/delius/service/CustodyServiceTest.java:51]: unrelated code warning fixes

File: src/test/java/uk/gov/justice/digital/delius/service/CustodyServiceTest.java [hunk 2]
```diff
@@ -549,4 +554,164 @@ public void willCreateTelemetryEventStatingNewOneHasBeenInserted() {
...
+        class WhenHasMultipleActiveEvents {
+            @BeforeEach
+            void setup() throws ConvictionService.DuplicateActiveCustodialConvictionsException {
+                when(offenderRepository.findByNomsNumber(anyString())).thenReturn(Optional.of(Offender.builder().offenderId(99L).build()));
+                when(convictionService.getSingleActiveConvictionByOffenderIdAndPrisonBookingNumber(anyLong(), anyString()))
+                        .thenThrow(new ConvictionService.DuplicateActiveCustodialConvictionsException(2));
+            }
+
+            @Test
+            @DisplayName("then a NotFoundException will be thrown")
```
Comment (mikehalmamoj) [inline src/test/java/uk/gov/justice/digital/delius/service/CustodyServiceTest.java:572]: I really like this nesting of display names to get nice descriptions of the tests.  It's a real shame that neither Intellij (ignoring @DisplayName) and Gradle (only shows last 2 levels of nesting) are not handling this very well.

**Review by mikehalmamoj — APPROVED** [review]
